data/src/emulator.js

```diff
@@ -809,17 +810,19 @@ class EmulatorJS {
             this.textElem.innerText = this.localization("Download Game Data");
 
             const gotGameData = (data) => {
-                if (["arcade", "mame"].includes(this.getCore(true))) {
-                    this.fileName = this.getBaseFileName(true);
+                const altName = this.getBaseFileName(true);
+
+                let ext = altName.split(".").pop().toLowerCase();
+                const coreName = this.getCore(true);
+                if (["arcade", "mame"].includes(coreName) || (coreName === "dos" && ext === "zip")) {
+                    this.fileName = altName;
```

Comment:
I don't think it should be done this way:

EmulatorJS supports multiple archive formats. It checks if the file is archived by the header bytes, not the extension (we do not always have this)
Specifically whitelisting the `zip` extension will break anywhere you try to feed it a blob

I think we should simply revert this part of the code, and add "dos" to that initial arcade/mame check, if the goal is to never extract and let dosbox handle the file.

Reply:
I think one solution that may work is having the opt-in ability to not decompress the rom, similar to dontDecompressBios - this could also help for more than just dosbox for systems that support a certain archive format and the website owner would rather let the core handle it